controllers/nodehealthcheck_controller.go

```diff
@@ -135,7 +134,7 @@ func (r *NodeHealthCheckReconciler) fetchNodes(ctx context.Context, labelSelecto
 	return nodes.Items, err
 }
 
-func (r *NodeHealthCheckReconciler) checkNodesHealth(nodes []v1.Node, nhc remediationv1alpha1.NodeHealthCheck) ([]v1.Node, error) {
+func (r *NodeHealthCheckReconciler) checkNodesHealth(nodes []v1.Node, nhc *remediationv1alpha1.NodeHealthCheck) ([]v1.Node, error) {
```

Comment:
why is a reference needed no mutation is needed?
I tent to use reference for arguments only if mutation is needed. And with that I know those method will not change the state without reading the code.

Reply:
Basically for better performance.
I'd like to avoid copying big Objects

Reply:
Please explain what performance problem you see here? 
First I value most the program correctness and simplicity. Hence the choice to use value and not a pointer. If we have a performance problem I want to see numbers. Here it is really unlikely to see one because of values.
Also, values do not need to be garbage collected, so expect for the code being immune (relatively) to mutability mistakes, it doesn't require any garbage collection (with all implications).

Reply:
> I'd like to avoid copying big Objects

+1
the question though is when "big" begins 🤷🏼 
another argument might also be how it is done at other places, for consistency